com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/linking/LinkingService.java

```diff
@@ -176,7 +221,18 @@ protected EObject getSingleElement(final EObject context, final EReference refer
     } else {
       desc = scope.getSingleElement(qualifiedLinkName);
     }
-    return desc == null ? null : desc.getEObjectOrProxy();
+
+    EObject target = desc == null ? null : desc.getEObjectOrProxy();
+
+    if (target != null && isImportRequired(context, reference, target)) {
+      if (target.eIsProxy()) {

```

Comment:
do you know in which cases it is proxy and in which not?

Reply:
My understanding is that whenever we use an index lookup to find our object, we have a proxy, since we get it from the index the name of the proxy is the one which has the calculated name using the logic in the export file.
For the cases where we look within a resource (e.g. some chained feature calls), then we have resolved objects.